pkg/sqlite/load_test.go

```diff
@@ -154,6 +155,71 @@ func TestAddPackageChannels(t *testing.T) {
 	}
 }
 
+func TestAddBundleSemver(t *testing.T) {
+	// Create a test DB
+	db, cleanup := CreateTestDb(t)
+	defer cleanup()
+	store, err := NewSQLLiteLoader(db)
+	require.NoError(t, err)
+	err = store.Migrate(context.TODO())
+	require.NoError(t, err)
+	graphLoader, err := NewSQLGraphLoaderFromDB(db)
+	require.NoError(t, err)
+
+	// Seed the db with a replaces-mode bundle/package
+	replacesBundle := newBundle(t, "csv-a", "pkg-foo", []string{"stable"}, newUnstructuredCSV(t, "csv-a", ""))
+	err = store.AddOperatorBundle(replacesBundle)
+	require.NoError(t, err)
+
+	err = store.AddPackageChannels(registry.PackageManifest{
+		PackageName: "pkg-foo",
+		Channels: []registry.PackageChannel{
+			{
+				Name:           "stable",
+				CurrentCSVName: "csv-a",
+			},
+		},
+		DefaultChannelName: "stable",
+	})
+	require.NoError(t, err)
+
+	// Add semver bundles in non-semver order.
+	bundles := []*registry.Bundle{
+		newBundle(t, "csv-3", "pkg-0", []string{"stable"}, newUnstructuredCSVWithVersion(t, "csv-3", "0.3.0")),
+		newBundle(t, "csv-1", "pkg-0", []string{"stable"}, newUnstructuredCSVWithVersion(t, "csv-1", "0.1.0")),
+		newBundle(t, "csv-2", "pkg-0", []string{"stable"}, newUnstructuredCSVWithVersion(t, "csv-2", "0.2.0")),
+	}
+	for _, b := range bundles {
+		graph, err := graphLoader.Generate(b.Package)
+		require.Conditionf(t, func() bool {
+			return err == nil || errors.Is(err, registry.ErrPackageNotInDatabase)
+		}, "got unexpected error: %v", err)
+		bundleLoader := registry.BundleGraphLoader{}
+		updatedGraph, err := bundleLoader.AddBundleToGraph(b, graph, &registry.AnnotationsFile{Annotations: *b.Annotations}, false)
+		require.NoError(t, err)
+		err = store.AddBundleSemver(updatedGraph, b)
+		require.NoError(t, err)
+	}
```

Comment:
I don't love that all of this code is in the test. It seems like this should be better encapsulated in non-test code. I suppose I could re-write the test using the [DirectoryPopulator](https://github.com/operator-framework/operator-registry/blob/28aa6545b5c24c1674b9a09194f9df7b9efb8031/pkg/registry/populator.go#L41) abstraction, but that would require using the `imageDirMap` and putting a bunch of bundle directories to disk during the test.

We could also look at making [this](https://github.com/operator-framework/operator-registry/blob/28aa6545b5c24c1674b9a09194f9df7b9efb8031/pkg/registry/populator.go#L170-L175) something that could be more easily called in a test.

I'm just not sure it's worth it given all of this code is deprecated.